aiven_mysql_migrate/utils.py

```diff
@@ -172,10 +180,63 @@ def sql_params(self):
         return params
 
 
-class MySQLDumpProcessor:
+class DumpProcessor(metaclass=ABCMeta):
     def __init__(self):
-        self.gtid = None
         self.gtid_block = ""
+        self.gtid = None
+
+    @abstractmethod
+    def process_line(self, line):
+        pass
+
+    def get_gtid(self):
+        return self.gtid
+
+
+class MydumperDumpProcessor(DumpProcessor):
+    def __init__(self, dump_output_dir: Path, backup_dir: Path):
+        super().__init__()
+        self.dump_output_dir = dump_output_dir
+        self.backup_dir = backup_dir
+
+    def process_line(self, line: str) -> str:
+        # metadata file is ready
+        if line and line.startswith("-- metadata "):
+            self._backup_metadata_file(line)
+        return line
+
+    def _backup_metadata_file(self, line: str) -> None:
+        """
+        Backup metadata file when mydumper indicates it's ready.
+
+        Args:
+            line: Line from mydumper output in format "-- <filename> <number>"
+        """
+        source_file = self.dump_output_dir / "metadata"
+        assert source_file.exists(), "Metadata file not found in dump output directory"
+
+        self.backup_dir.mkdir(parents=True, exist_ok=True)
+
+        dest_file = self.backup_dir / "metadata"
+        shutil.copy2(source_file, dest_file)
+        self.gtid = self._extract_gtid_from_metadata()
```

Comment:
As MydumperDumpProcessor instance lives for the whole migration we can just take GTID from the source file and don't copy it, we don't do any restart recovery anyway, there is no reason to keep it other than just in case...